### PR TITLE
util.cc: reduce two read_file functions into one

### PR DESCRIPTION
**Pull request by deanlee**
`read_file` is only used here:
https://github.com/commaai/openpilot/blob/91504176e403ca1bb6bd9491ce9e2075b7f4bae0/selfdrive/modeld/runners/snpemodel.cc#L28

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Can we just use the other read_file instead of having two different implementations?

**Comment by deanlee** [comment]
`util::read_file `is not suitable to read large file because it has redundant copies and many resize operations during read large file. it's almost 10x slower than read_file (tested on EON):
```
util::read_file("supercombo.dlc") : 2599 ms
read_file("supercombo.dlc", &out_size) : 260 ms
```

I suggest to keep `read_file` and optimize `util::read_file `later

**Comment by pd0wm** [comment]
Can we just optimize it now? Is it fast if you reserve the correct size before reading?

**Comment by deanlee** [comment]
updated. This implementation is almost as fast as the C version.

File: selfdrive/common/util.cc [hunk 1]
```diff
@@ -84,3 +54,19 @@ int set_core_affinity(int core) {
...
 #endif
 }
+
+namespace util {
+
+std::string read_file(const std::string& fn) {
+  std::ifstream ifs(fn, std::ios::binary | std::ios::ate);
+  if (ifs) {
+    std::ifstream::pos_type pos = ifs.tellg();
+    std::string result(pos, '\0');
```
Comment (pd0wm) [inline selfdrive/common/util.cc:64]: Why not use resize? Now you're writing the array twice.

File: selfdrive/common/util.cc [hunk 2]
```diff
@@ -84,3 +54,20 @@ int set_core_affinity(int core) {
   return -1;
 #endif
 }
+
+namespace util {
```
Comment (adeebshihadeh) [inline selfdrive/common/util.cc:58]: why is read_file in the namespace but not write_file?

File: selfdrive/common/util.cc [hunk 3]
```diff
@@ -84,3 +54,20 @@ int set_core_affinity(int core) {
...
+  std::ifstream ifs(fn, std::ios::binary | std::ios::ate);
+  if (ifs) {
+    std::ifstream::pos_type pos = ifs.tellg();
+    std::string result;
+    result.resize(pos);
+    ifs.seekg(0, std::ios::beg);
+    ifs.read(result.data(), pos);
+    if (ifs) return result;
+  }
+  return {};
```
Comment (adeebshihadeh) [inline selfdrive/common/util.cc:70]: why not? think this is more readable
```suggestion
  return "";
```

**Pull request merged**